exporter/sumologicexporter/config.go

```diff
@@ -0,0 +1,126 @@
+	//   * text - Logs will appear in Sumo Logic in text format.
+	//   * json - Logs will appear in Sumo Logic in json format.
+	LogFormat LogFormatType `mapstructure:"log_format"`
+
+	// Metrics related configuration
+	// The format of metrics you will be sending, either graphite or carbon2 or prometheus (Default is carbon2)
+	MetricFormat MetricFormatType `mapstructure:"metric_format"`
+
+	// List of regexes for attributes which should be send as metadata
+	MetadataAttributes []string `mapstructure:"metadata_attributes"`
+
+	// Sumo specific options
+	// Desired source category.
+	// Useful if you want to override the source category configured for the source.
+	SourceCategory string `mapstructure:"source_category"`
+	// Desired source name.
+	// Useful if you want to override the source name configured for the source.
+	SourceName string `mapstructure:"source_name"`
+	// Desired host name.
+	// Useful if you want to override the source host configured for the source.
+	SourceHost string `mapstructure:"source_host"`
+	// Name of the client
+	Client string `mapstructure:"client"`
+}
+
+// CreateDefaultHTTPClientSettings returns default http client settings
+func CreateDefaultHTTPClientSettings() confighttp.HTTPClientSettings {
+	return confighttp.HTTPClientSettings{
+		Timeout: defaultTimeout,
+	}
+}
+
+// LogFormatType represents log_format
+type LogFormatType string
+
+// MetricFormatType represents metric_format
+type MetricFormatType string
+
+// PipelineType represents type of the pipeline
+type PipelineType string
+
+// CompressEncodingType represents type of the pipeline
+type CompressEncodingType string
+
+const (
+	// TextFormat represents log_format: text
+	TextFormat LogFormatType = "text"
+	// JSONFormat represents log_format: json
+	JSONFormat LogFormatType = "json"
+	// GraphiteFormat represents metric_format: text
+	GraphiteFormat MetricFormatType = "graphite"
+	// Carbon2Format represents metric_format: json
+	Carbon2Format MetricFormatType = "carbon2"
+	// PrometheusFormat represents metric_format: json
+	PrometheusFormat MetricFormatType = "prometheus"
+	// GZIPCompression represents compress_encoding: gzip
+	GZIPCompression CompressEncodingType = "gzip"
+	// DeflateCompression represents compress_encoding: deflate
+	DeflateCompression CompressEncodingType = "deflate"
+	// NoCompression represents disabled compression
+	NoCompression CompressEncodingType = ""
+	// MetricsPipeline represents metrics pipeline
+	MetricsPipeline PipelineType = "metrics"
+	// LogsPipeline represents metrics pipeline
+	LogsPipeline PipelineType = "logs"
+	// defaultTimeout
+	defaultTimeout time.Duration = 55 * time.Second
+	// DefaultCompress defines default Compress
+	DefaultCompress bool = true
+	// DefaultCompressEncoding defines default CompressEncoding
+	DefaultCompressEncoding CompressEncodingType = "gzip"
+	// DefaultMaxRequestBodySize defines default MaxRequestBodySize in bytes
+	DefaultMaxRequestBodySize int = 20 * 1024 * 1024
```

Comment:
Isn't this related to `max_request_body_size`? The readme says the default is 1MB. Or is the recommended default different than the actual default value?

Reply:
Readme contains valid value